### DISCUSSION TITLE
What does "Create Symlink" do exactly?

### DISCUSSION THREAD

**Discussion by Xiarno** [Victor-IX/Blender-Launcher-V2#98] [Q&A] [answered]
I just recently learned what a Symlink is, so I know the basics of it. But when right-clicking on a build, you can "Create Symlink". What does it do exactly? Where does it put it too? Thanks!

**Comment by Victor-IX** [comment]
I am not an expert on how this works inside of the Blender Launcher, but when you create a symlink for a Blender version it creates a new folder bl_simlink in the Blender Launcher library

**Comment by eisenbruch** [comment]
In the example of Blender Launcher, what you can do with it is create a symlink to a master blender folder with your addons. 
this way you can run multiple versions of blender, all with the same addons installed, without needing to do anything manually.
if you need to see how this works, I can prob record a video of it from my machine later. easiest to show rather than explain

**Reply by Victor-IX** [reply]
Oh yes! I will be interested to see how you have that setup

**Reply by Xiarno** [reply]
Hey!
I'm interested in how you set it up like you said. Because when I right click a build in the Blender Launcher and click "Create Symlink", I do not get to choose anything. So I'm not sure.

Also, wouldn't that sometime create issues when the addon is incompatible with certains version? I know some addons that the latest version is incompatible with an earlier version or just not compatible with the newest version yet.

**Reply by eisenbruch** [reply, marked as answer]
@Victor-IX @Xiarno  here you guys go:

https://github.com/Victor-IX/Blender-Launcher-V2/assets/6180294/f73af167-26e0-42be-87f9-ab3ae7790411

Sorry its a bit long winded but I wanted to be thorough.

Any questions, feel free to ask.

**Reply by Xiarno** [reply]
The "software" you are talking about is the one I recently found.
Here it is: https://schinagl.priv.at/nt/hardlinkshellext/linkshellextension.html

It seems to be exactly the same as you are using

---

Also, while it doesn't exactly answers my original question on how do you exactly use the symlinks from the Blender Launcher (Aka what is the purpose of the "Create symlinks" button), I'll still take it as a correct answers. But feel free to add more onto it! Especially regardings assets. Addons are cool & all, but I wonder what you could do with assets.

**Reply by eisenbruch** [reply]
another video!

https://github.com/Victor-IX/Blender-Launcher-V2/assets/6180294/fc594fc3-ae2f-413f-9fcf-7dc9ddbdf296

it creates a symbolic link to the blender install directory in your blender launcher master or library folder.

i dont know why that is helpful.

probably someone elses reasons!

this seems to be for differnt purposes than what I was explaining

**Reply by Xiarno** [reply]
Yeah that's what I also thought

**Reply by eisenbruch** [reply]
while we're in that menu. where can i find info on Register Extension and Install Template?

**Reply by Xiarno** [reply]
Idk about install template, but I'm pretty sure "Register Extension" just means that every .blend file will open whatever is registered.

**Reply by eisenbruch** [reply]
ah. doesnt seem to work for me. windows still asks what program to use whenever i double click a blender file. im fine with that tho.

**Reply by Xiarno** [reply]
Yeah :/
Just go into Blender, Preferences and do it directly from here.
Or just go into "Defaults Apps" settings in Windows 10 and change it manually from there.

**Reply by Victor-IX** [reply]
Thanks for the video, nicely explained! 
Honestly, I don't really know the use case for the current symlink option, the doc doesn't explain its use. 
We plan to implement a custom solution to simplify addons and preference management inside the blender launcher but I want to wait a bit for the Blender Extention project to be released as it will change a lot how addons are managed

**Reply by Xiarno** [reply]
What is the Blender extension project?

**Reply by Victor-IX** [reply]
https://extensions.blender.org
https://devtalk.blender.org/t/extensions-platform-alpha-launch/33342

The extension project is the new project that will change how addons are installed and managed inside Blender, it allows anybody to add "built-in" addons inside Blender it also adds an addon updater inside Blender and a lot more

**Reply by zeptofine** [reply]
from what I know, the blender extension project is an official method to host, fetch, distribute and update blender add-ons and themes. think blendermarket and gumroad. it also makes a lot of changes to addon structure. you can see it here https://extensions.blender.org/. you can try it out by activating it on daily builds with the developer experimental settings

**Reply by Xiarno** [reply]
Oooo! That's awesome!
So one day it would replace the Blendermarket?

**Reply by zeptofine** [reply]
not necessarily because it doesn't really allow paid products I'm pretty sure but they're comparable for free add-ons I think

**Reply by Xiarno** [reply]
Ah yes I see you are right.
